### PR TITLE
Auto port selection and SunnyWeatherSpec for Artery

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

File: akka-cluster/src/multi-jvm/scala/akka/cluster/artery/ArterySunnyWeatherSpec.scala [hunk 1]
```diff
@@ -1,18 +1,19 @@
 /**
  * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
  */
-package akka.cluster
+package akka.cluster.artery
```
Comment (patriknw) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/artery/ArterySunnyWeatherSpec.scala]: we probably want to keep both version as long as we support the old remoting, so a copy, or make the effort to do 2 variants of the test with 2 different configs. I think we have that in some tests already, such as those with the failure detector puppet, e.g. https://github.com/akka/akka/blob/master/akka-cluster/src/multi-jvm/scala/akka/cluster/ClientDowningNodeThatIsUpSpec.scala

File: akka-cluster/src/multi-jvm/scala/akka/cluster/artery/ArterySunnyWeatherSpec.scala [hunk 2]
```diff
@@ -24,12 +25,25 @@ object SunnyWeatherMultiJvmSpec extends MultiNodeConfig {
...
-    akka.actor.provider = akka.cluster.ClusterActorRefProvider
-    akka.loggers = ["akka.testkit.TestEventListener"]
-    akka.loglevel = INFO
-    akka.remote.log-remote-lifecycle-events = off
-    akka.cluster.failure-detector.monitored-by-nr-of-members = 3
+    akka {
+      actor {
+        provider = akka.cluster.ClusterActorRefProvider
+        serialize-creators = false
+        serialize-messages = false
```
Comment (patriknw) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/artery/ArterySunnyWeatherSpec.scala]: that should not be needed, I hope
I think I disabled that for performance tests, for performance reasons.

File: akka-cluster/src/multi-jvm/scala/akka/cluster/artery/ArterySunnyWeatherSpec.scala (hunk 1 above)
Comment (johanandren) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/artery/ArterySunnyWeatherSpec.scala]: Äsch. My intention was to copy. Will try to make it a config instead now that I know that I didn't have to change anything but the config.

File: akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala [hunk 3]
```diff
@@ -352,3 +359,19 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
...
 }
 
+object ArteryTransport {
+
+  /**
+   * Internal API
+   * @param hostname
+   * @return A port that is hopefully available
+   */
+  private[artery] def autoSelectPort(hostname: String): Int = {
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: we are mixing a bit with the visibility, I think we should decide on `private[remote]` as the default if no other reason
I'm to blame for several `private[akka]`

**Comment by patriknw** [comment]
Yay!

File: akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala (hunk 3 above)
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: Went with as narrow as possible but still testable, but I'll change, no problem.

File: akka-cluster/src/multi-jvm/scala/akka/cluster/SunnyWeatherSpec.scala [hunk 4]
```diff
@@ -14,35 +14,61 @@ import scala.collection.immutable.SortedSet
...
+      remote.log-remote-lifecycle-events = off
+      cluster.failure-detector.monitored-by-nr-of-members = 3
+    }
+    """
+
+  val arteryConfig =
+    """
+    akka.remote.artery {
+      enabled = on
+      port = 0
```
Comment (patriknw) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/SunnyWeatherSpec.scala]: look at `MultiNodeSpec`, it sets `"akka.remote.netty.tcp.port" -> selfPort` (and `selfPort` is by default 0)
I think we should amend, `MultiNodeSpec` to do the same with `akka.remote.artery.port`, and then this is not needed here.

**Comment by ktoso** [comment]
LGTM AFAICS

File: akka-cluster/src/multi-jvm/scala/akka/cluster/SunnyWeatherSpec.scala [hunk 5]
```diff
@@ -14,35 +14,60 @@ import scala.collection.immutable.SortedSet
...
-class SunnyWeatherMultiJvmNode3 extends SunnyWeatherSpec
-class SunnyWeatherMultiJvmNode4 extends SunnyWeatherSpec
-class SunnyWeatherMultiJvmNode5 extends SunnyWeatherSpec
+class SunnyWeatherRemotingMultiJvmNode1 extends SunnyWeatherRemotingSpec
+class SunnyWeatherRemotingMultiJvmNode2 extends SunnyWeatherRemotingSpec
+class SunnyWeatherRemotingMultiJvmNode3 extends SunnyWeatherRemotingSpec
+class SunnyWeatherRemotingMultiJvmNode4 extends SunnyWeatherRemotingSpec
+class SunnyWeatherRemotingMultiJvmNode5 extends SunnyWeatherRemotingSpec
+
+class SunnyWeatherArteryMultiJvmNode1 extends SunnyWeatherRemotingSpec
```
Comment (ktoso) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/SunnyWeatherSpec.scala]: Shouldn't this extend `SunnyWeatherArterySpec`?

Comment (johanandren) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/SunnyWeatherSpec.scala]: indeed

**Comment by johanandren** [comment]
PLS BUILD

**Comment by patriknw** [comment]
LGTM

**Pull request merged**